src-ts/lib/work-provider/work-functions/work-factory/work.factory.ts

```diff
@@ -289,3 +289,77 @@ function getTypeCategory(type: WorkType): WorkTypeCategory {
             return WorkTypeCategory.unknown
     }
 }
+
+export function mapFormData(type: string, formData: any): any {
+    switch (type) {
+        case (WorkType.problem):
+            return buildFormDataProblem(formData)
+        case (WorkType.data):
+            return buildFormDataData(formData)
+        case (WorkType.findData):
+            return buildFormDataFindData(formData)
+        case (WorkType.design):
+            return buildFormDataDesign(formData)
+        default:
+            return formData
+    }
+}
+
+function buildFormDataProblem(formData: any): any {
+    return {
+        projectTitle: formData.projectTitle,
+        // tslint:disable-next-line: object-literal-sort-keys
+        goal: { title: 'What\'s Your Goal?', value: formData.goals?.value },
+        data: {
+            title: 'What Data Do You Have?', value: [
+                formData.sampleData?.value,
+                formData.assetsDescription?.value,
+            ],
+        },
+    }
+}
+
+function buildFormDataData(formData: any): any {
+    return {
+        projectTitle: formData.projectTitle,
+        // tslint:disable-next-line: object-literal-sort-keys
+        data: { title: 'Share Your Data (Optional)', value: formData.assetsUrl?.value },
```

Comment:
in general, i'd prefer we list each key on its own line w/trailing commas. i should prob find a lint rule that will do this on save.